src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/GitHookTasks.kt

```diff
@@ -33,13 +33,20 @@ open class InstallPrePushHookTask : InstallHookTask("pre-push") {
  * Install or update a kotlinter-gradle hook.
  */
 abstract class InstallHookTask(@get:Internal val hookFileName: String) : DefaultTask() {
-    @Input
+
+    @get:Input
     val gitDirPath: Property<String> = project.objects.property(default = ".git")
 
     @get:Internal
     abstract val hookContent: String
 
+    @get:Input
+    protected abstract val rootProjectDir: Property<File>
```

Comment:
This would lead to a:
```
Caused by: org.gradle.internal.execution.fingerprint.InputFingerprinter$InputFingerprintingException: Cannot fingerprint input property 'rootProjectDir': value '/private/var/folders/mf/09fz6s0n6574_rz1fw95581r0000gn/T/junit8185502116311246174' cannot be serialized.
	at org.gradle.internal.execution.fingerprint.impl.DefaultInputFingerprinter$InputCollectingVisitor.visitInputProperty(DefaultInputFingerprinter.java:117)
	at org.gradle.api.internal.tasks.execution.TaskExecution.visitRegularInputs(TaskExecution.java:315)
	at org.gradle.internal.execution.fingerprint.impl.DefaultInputFingerprinter.fingerprintInputProperties(DefaultInputFingerprinter.java:61)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.captureExecutionStateWithOutputs(CaptureStateBeforeExecutionStep.java:193)
	at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.lambda$captureExecutionState$1(CaptureStateBeforeExecutionStep.java:141)

```

Reply:
I looked into the exception and it turns out:
1. We can't use `RegularFileProperty` with `@Input` annotation. `RegularFileProperty` should be annotated with `InputFile`. 
2. Secondly, `RegularFileProperty` isn't meant to be used with directories, which the `project.rootProject.rootDir` value is
3. To _properly_ observe a directory input it's required to use `DirectoryProperty` + `@InputDirectory`. 
4. However, in this case, we don't really want to observe `rootProjectDir`'s content. We only want to know the file path, not the file content. 
... which answers my initial question :) 

Given all the above, I'd still propose to keep the new property consistent with others, avoid `LeakingThis` and have a concise single-line expression:

```suggestion
    @Input
    val rootProjectDir = project.objects.property(default = project.rootProject.rootDir)
```

Reply:
These are interesting observations, thanks for sharing!
That part of the gradle docs always seem very unclear to me.